types/xdg_shell/wlr_xdg_surface.c

```diff
@@ -81,14 +81,13 @@ void unmap_xdg_surface(struct wlr_xdg_surface *surface) {
 	}
 
 	surface->configured = surface->mapped = false;
-	surface->configure_serial = 0;
 	if (surface->configure_idle) {
 		wl_event_source_remove(surface->configure_idle);
 		surface->configure_idle = NULL;
 	}
 	surface->configure_next_serial = 0;
 
-	memset(&surface->geometry, 0, sizeof(struct wlr_box));
+	surface->current = surface->pending;
 	memset(&surface->pending, 0, sizeof(struct wlr_xdg_surface_state));
```

Comment:
>Once the window geometry of the surface is set, it is not possible to unset it, and it will remain the same until set_window_geometry is called again, even if a new subsurface or buffer is attached.

I don't think the state should be cleared here. Not sure why was this done in the first place though.

Reply:
Ah, good catch.

Reply:
Hm, wait, I think I mixed things up. This is the unmap handler, it needs to reset all of the state.

Fixed.